tron/core/actiongraph.py

```diff
@@ -19,59 +14,38 @@ def __init__(self, graph, action_map):
         self.action_map = action_map
 
     @classmethod
-    def from_config(cls, actions_config, cleanup_action_config=None):
+    def from_config(cls, action_map, config_context, cleanup_action=None):
         """Create this graph from a job config."""
-        actions = {
-            maybe_decode(name): action.Action.from_config(conf)
-            for name, conf in six.iteritems(actions_config)
-        }
-        if cleanup_action_config:
-            cleanup_action = action.Action.from_config(cleanup_action_config)
-            actions[maybe_decode(cleanup_action.name)] = cleanup_action
+        if cleanup_action:
+            action_map = action_map.set(cleanup_action.name, cleanup_action)
 
-        return cls(cls._build_dag(actions, actions_config), actions)
+        return cls(*cls._build_dag(action_map))
```

Comment:
This unpacking is "tricky" to me, but I see how it is cleaner than the previous method